### PR TITLE
Aman 87/delete certif

### PR DESCRIPTION
**Pull request by Scarletra**
# Description  
Add new feature: delete certification on certification section

# Ticket Issue 
[[Provide link to Jira]  ](https://a-man-ppl.atlassian.net/browse/AMAN-87?atlOrigin=eyJpIjoiYmY0OWEzOTI5MGMyNDIzOGEzNTVmZDI5MDE5N2Q5YzciLCJwIjoiaiJ9)

# Changes Made  
Add new button 'Hapus' and delete functionality on certification section

# Issue Type
- [ ] 🐛 Bug fix 
- [ ] ⚙️ Refactor
- [X] 💻 New feature 
- [ ] 📄 Documentation update
- [ ] ⚠️ Breaking changes
- [X] 🔍 Testing

# Added/updated tests?

- [X] Yes
- [ ] No, and this is why: [please replace this line with details on why tests have not been included]

# Additional Notes  
You can try this feature while editing one of the certification

# (optional) What photo/gif best describes this PR or how it makes you feel?
No photo or gif for now

## Summary by CodeRabbit

- **New Features**
  - Introduced certificate deletion functionality, enabling users to remove certificates directly from the management interface.
  - Enhanced the certificate editing experience with an improved modal layout for streamlined actions.
  - Added a new custom hook for managing certification data fetching and state.
  - Implemented a new `ModalFormWrapper` component for improved modal handling.

- **Refactor**
  - Upgraded certificate data handling with a new class-based structure for managing certification services.

- **Bug Fixes**
  - Resolved duplicate navigation entry issues to ensure smoother access and stability.

- **Tests**
  - Added comprehensive test coverage for the certificate management component, including positive, negative, and edge cases.
  - Updated existing tests to reflect changes in the output format for certificate titles and file sizes.

### CODE REVIEW

File: src/services/CertificationService.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }
+
+  static async addCertificate(token: string, certificateData: CertificateDetail) {
+    try {
+      const formData = new FormData();
+      formData.append('title', certificateData.title);
+      
+      if (certificateData.publishDate) {
+        formData.append('publishDate', certificateData.publishDate);
+      }
```
[the remark this answers is not in the document]
Comment (Scarletra) [reply src/services/CertificationService.ts:31]: buat yg url endpoint udh solved ya

[the remark this answers is not in the document]
Comment (Scarletra) [reply src/services/CertificationService.ts]: buat yg ganti impl tanpa publish date solved jg yaa, siap merge
